### ISSUE TITLE
FallThrough module does not support the spelling "fall-through" by default

### ISSUE THREAD

**Issue by guillaumelambert** [checkstyle/checkstyle#7374]
The FallThrough module supports by default various spellings of the word fallthrough in comments.
These spellings are listed at https://checkstyle.sourceforge.io/config_coding.html or https://checkstyle.org/config_coding.html#FallThrough

Although, one of the main common spelling "fall-through" is not supported 
According to wiktionary, fallthrough is the alternate spelling of fall-through which is the primary form.
https://en.wiktionary.org/wiki/fallthrough
https://en.wiktionary.org/wiki/fall-through
Note also this spelling is used in the description of the documentation of the module...

NB: ticket opened in JIRA previously https://issues.apache.org/jira/browse/MCHECKSTYLE-383?page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel&focusedCommentId=16994795#comment-16994795

```
$ cd/tmp/
$ mkdir example
$ cd example
$ wget https://github.com/checkstyle/checkstyle/releases/download/checkstyle-8.27/checkstyle-8.27-all.jar
$ cat >config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">


<module name = "Checker">
    <module name="TreeWalker">
		<module name="FallThrough"/>
    </module>
</module>
^C
$ RUN_LOCALE="-Duser.language=en -Duser.country=US"
$ cat >example.java 
public class example {
// adapted from the original example at https://checkstyle.org/config_coding.html#FallThrough

    public static void main(String[] args) {
	int i=0;
	switch (i) {
		case 0:
		  i++;
		// fall-through
		case 1:
		  i++;
		  // falls through
		case 2:
		case 3:
		case 4: {
		  i++;
		}
		// fallthrough
		case 5:
		  i++;
		/* fallthru */case 6:
		i++;
		// note that the trailing semicolon is missing in the original example at https://checkstyle.org/config_coding.html#FallThrough
		break;
	}
        System.out.println(i);
    }

}
^C
$ java $RUN_LOCALE  -jar checkstyle-8.27-all.jar  -c config.xml example.java 
Starting audit...
[ERROR] /tmp/checkstyle/example.java:10:17: Fall through from previous branch of the switch statement. [FallThrough]
Audit done.
Checkstyle ends with 1 errors.
```

---------------

No error should be raised. The comment "fall-through" should be supported by default since it is a correct spelling contrary to "fallthru" (which is supported).
In other words, the default ReliefPattern regex should  be "fallthru|falls?[ -]?through" instead of "fallthru|falls? ?through"

--------------

Default Regexp is changed. 
Aditionally become relief "fall thru".

**Comment by romani** [comment]
@guillaumelambert , please stricktly follow issue template and you will figure out yourself how to resolve issue on your side.

**Comment by guillaumelambert** [comment]
> 
> 
> @guillaumelambert , please stricktly follow issue template and you will figure out yourself how to resolve issue on your side.

OK I will try to do that tomorrow.
Note that I posted the issue on the official project because the incorrect slang spelling "fallthru" is supported by default...

**Comment by guillaumelambert** [comment]
@romani I've updated the Issue description. Is it OK like that ?

**Comment by romani** [comment]
Ok, you want it to be by default.
https://checkstyle.org/styleguides/google-java-style-20180523/javaguide.html#s4.8.4.2-switch-fall-through mention it as `fall through` but in text it is `fall-through`.

@rnveach , do you think it is reasonable to update default regexp?

**Comment by rnveach** [comment]
I am fine with this to allow `fall-through` as well as the other variations.

**Comment by romani** [comment]
@guillaumelambert , the quickest way to get it, is do contributions yourself, please be welcome with PR, we will guide on how to make contribution properly.

**Comment by guillaumelambert** [comment]
@romani fair enough for me, I will try to have a look at your doc to see how to make PR and I'll propose you something as soon that I have some free cycles.

**Comment by guillaumelambert** [comment]
@romani
As explained at https://checkstyle.org/beginning_development.html, I've forked the project and push a working branch on my new repo.
The diff is visible at this URL https://github.com/guillaumelambert/checkstyle/commit/ea89f5d3130491d10c6e811ca21561f7dea1d9d2
The next instruction is to wait for a mentor's feedback before creating a Pull Request.

About this proposal, I modified the regex and tried to update the examples, documentations and comments. ( I fixed by the way the trailing semicolon missing in the original example of the documentation. )
I used the initial regex proposed here (i.e. "fallthru|falls?[ -]?through" instead of "fallthru|falls? ?through" ) but I wonder now if the regex "falls?[ -]?thr(u|ough)" wouldn't make more sense.
I haven't also modified any Junit test or test file for this new spelling. I am not sure this is needed nor required.

**Comment by romani** [comment]
>I wonder now if the regex "falls?[ -]?thr(u|ough)" wouldn't make more sense.

I am ok with this.

Please create PR and we'll start acceptance process. As you add just another allowance word, please append it existing cases (do not put it in the middle that cause extra diff and that will complicated review)

**Comment by guillaumelambert** [comment]
@romani Thanks for the feedback.
I've updated the commit accordingly and proposed a pull request.

**Comment by romani** [comment]
fix is merged.
